docs/spec/lcd/api.md

```diff
@@ -0,0 +1,847 @@
+    "result": {
+      "id": "992e24f5761b37de48536cecff0a0687937049a3",
+            "listen_addr": "10.0.2.15:46656",
+            "network": "test-chain-F0bln0",
+            "version": "0.19.7-dev",
+            "channels": "4020212223303800",
+            "moniker": "lhy-ubuntu",
+            "other": [
+                "amino_version=0.9.9",
+                "p2p_version=0.5.0",
+                "consensus_version=v1/0.2.2",
+                "rpc_version=0.7.0/3",
+                "tx_index=on",
+                "rpc_addr=tcp://0.0.0.0:46657"
+      ]
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot get the node info",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /syncing - GET
+
+url: /syncing, Method: GET
+
+Functionality: Check syncing status of the fullnode which is connecting with the LCD node
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "syncing":false
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot get the syncing info",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+### /blocks/latest - GET
+
+url: /blocks/latest, Method: GET
+
+Functionality: Get the lasted block and verify it
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "authentic":true
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot verify the latest block",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /blocks/{height} - GET
+
+url: /blocks/{height}, Method: GET
+
+Functionality: Get the block at specified height and verify it
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "authentic":true
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot verify the block at this height",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /validatorsets/latest - GET
+
+url: /validatorsets/latest, Method: GET
+
+Functionality: Get the lasted validatorsets and verify it
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "authentic":true
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot verify the latest block",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /validatorsets/{height} - GET
```

Comment:
why `/validatorsets` instead of simply `validators`?

Reply:
Because the resource `validators` would implicate, that an entity in the resource is one validator. But the endpoint returns validator sets at a certain height.

Reply:
what is a validator set, if not a bunch of validators?

Reply:
But we have only one validator set at a given height?

Reply:
Yes we only have one validator set at a given height.
A `validatorset` is a bunch of validators: in code of array validators.
The resource `validatorsets` holds an array of array of validators.
If you select from `validatorsets` you get an array.
If you would select from `validators` you would expect an object (one validator).

Reply:
would it make sense to use a dash? `validator-set`?

Reply:
that would be an option. or use an underscore so it is completely selected on double-click: `validator_sets`.